tensorflow/compiler/tf2tensorrt/convert/ops/tile.cc

```diff
@@ -0,0 +1,169 @@
+    CHECK_INPUT_SIZE(inputs.size(), 2, params.node_def);
+
+    const auto &repl = inputs.at(1);
+    if (params.use_implicit_batch && repl.is_tensor()) {
+      return errors::InvalidArgument("Conversion for Tile is not implementd for multipliers passed as a tensor");
+    }
+
+    nvinfer1::DataType dtype;
+    const int *multiplies;
+    if (repl.is_weights()) {
+      CHECK_SHAPE_TENSOR(repl.weights().GetTensor());
+      dtype = repl.weights().TrtDType();
+      multiplies = repl.weights().GetPointer<int>();
+    } else {
+      dtype = repl.tensor()->getType();
+      multiplies = nullptr;
+    }
+
+    if (dtype != nvinfer1::DataType::kINT32) {
+      return errors::InvalidArgument("The replication parameter of the ", params.node_def.op(), " operation in ",
+                                     params.node_def.name(), " is expected to be of ",
+                                     DebugString(nvinfer1::DataType::kINT32),
+                                     " type, got ", DebugString(dtype));
+    }
+
+    const auto dims = inputs.at(0).GetTrtDims();
+    const auto nb_dims = dims.nbDims + (params.use_implicit_batch && inputs.at(0).is_tensor()? 1 : 0);
+    if (multiplies) {
+      const int mult_numb = repl.weights().count();
+      if (mult_numb != nb_dims) {
+        return errors::InvalidArgument("The lenght of the replication vector (", mult_numb,
+                                       ") of the Tile operation in '", params.node_def.name(),
+                                       "' is expected to be equal to the number of dimentions of the input vector (",
+                                       nb_dims, ")");
+      }
+
+      bool positiveReplic = true;
+      for (int i = 0; i < nb_dims; i++)
+        positiveReplic &= multiplies[i] > 0;
+
+      if (!positiveReplic) {
+        char buffer[256], *pBuff = buffer;
+        for (int i = 0; i < nb_dims; i++)
+          pBuff += sprintf(pBuff, (i? ", %d" : "%d"), multiplies[i]);
+
+         return errors::InvalidArgument("All replications of the Tile operation in ", params.node_def.name(),
+                                     " should be positive, got (", buffer, ").");
+      }
+    }
+
+    return Status::OK();
+  }
+
+  Status Convert() {
+    const auto &params = *this->params_;
+    const auto& inputs = params.inputs;
+    auto *converter = params.converter;
+    auto *network = converter->network();
+    const auto &tensor = inputs.at(0);
+    const auto &replics = inputs.at(1);
+    const auto dims = tensor.GetTrtDims();
+    const auto nb_dims = dims.nbDims;
+    const auto dim_adj = params.use_implicit_batch && tensor.is_tensor()? 1 : 0;
+
+    bool dynamic_flag = false;
+    const int *pMultiplies = nullptr;
+    if (replics.is_tensor()) {
+      dynamic_flag = true;
+    } else {
+      pMultiplies = replics.weights().GetPointer<int>();
+    }
+
+    nvinfer1::Dims size{nb_dims, {1}};
+    nvinfer1::Dims stride{nb_dims, {1}};
+    const auto *pSize = dims.d;
```

Comment:
Use new DimsAdapter (defined in `tf2tensorrt/convert/utils.h`) on master top-of-tree rather than raw pointer on the dims struct.

Reply:
Sorry, I cannot find DimsAdapter in current version of `tf2tensorrt/convert/utils.h`

Reply:
You probably need to wait until the PR is merged to fix this. In the process of merging that PR.

Reply:
OK, I will wait this merge.

Reply:
Sorry, I'm not sure how efficient it is to use `DimsAdapter` here. As far as I understand, we still need to create some intermediate vectors and cut out the first coordinates. In the current version, we do this directly.

Reply:
I think you can still you DimAdapter to get "stride".
To compute dynamic_flag, can you use std::any_of?
Can you explain your expression "(size.d[i] = pMultiplies[dim_adj+i] * *(pSize+i)) < 0"?
Do you mean dynamic_flag is set to true if there is a pair in "size" and "pMultiplies", with the same negative values?

Reply:
1. Done: `dynamic_flag` calculated with `std::any_of`
2. dynamic_flag is set to true when `(pMultiplies[dim_adj+i] * *(pSize+i)) < 0`. 
    Actually,  `pMultiplies[dim_adj+i] > 0` (according to corresponding check  in `ConvertTile::Validate`), therefore what we check here is just `*(pSize+i) < 0` for all i.